lib/ansible/plugins/lookup/config.py

```diff
@@ -56,32 +71,81 @@
   type: raw
 """
 
+import ansible.plugins.loader as plugin_loader
+
 from ansible import constants as C
-from ansible.errors import AnsibleError
+from ansible.errors import AnsibleError, AnsibleLookupError, AnsibleOptionsError
+from ansible.module_utils._text import to_native
 from ansible.module_utils.six import string_types
 from ansible.plugins.lookup import LookupBase
+from ansible.utils.sentinel import Sentinel
+
+
+class MissingSetting(AnsibleError):
+    pass
```

Comment:
Should this handle formatting the traceback? The exception shows up as unhandled.

```
fatal: [localhost]: FAILED! => {
    "msg": "An unhandled exception occurred while running the lookup plugin 'config'. Error was a <class 'ansible.errors.AnsibleLookupError'>, original message: Unable to find setting aws_profile. Requested entry (plugin_type: inventory plugin: amazon.aws.aws_ec2 setting: aws_profile ) was not defined in configuration.. Requested entry (plugin_type: inventory plugin: amazon.aws.aws_ec2 setting: aws_profile ) was not defined in configuration."
}
```

Reply:
ah, yes, should inherit from AnsibleOptionsError instead